### DISCUSSION TITLE
Get nearest, paginated items using Prisma and MongoDB with 2dsphere Index

### DISCUSSION THREAD

**Discussion by casperibo** [prisma/prisma#11776] [Q&A]
Hi, I'm using Prisma MongoDB Nexus for my project and I try to list nearest items with pagination.

```
const test = await ctx.prisma.store.findRaw({
          filter: {
            location: {
              $nearSphere: {
                $geometry: {
                  type: "Point",
                  coordinates: [-73.93414657, 40.82302903],
                },
              },
            },
          },
        });
```

I can get nearest results but I can't use take,skip arguments like down below.

```
const test2 = await ctx.prisma.store.findMany({
          take: 10,
          skip: 50,
        });
```

I'm using graphql-relay for pagination by the way.

```
const offset = after ? cursorToOffset(after) + 1 : 0;
        if (isNaN(offset)) throw new Error("cursor is invalid");

        const [totalCount, items] = await Promise.all([
          ctx.prisma.store.count(),
          ctx.prisma.store.findMany({
            take: first,
            skip: offset,
          }),
        ]);

        return connectionFromArraySlice(
          items,
          { first, after },
          { sliceStart: offset, arrayLength: totalCount }
        );
```

Any help would be appreciated.

**Comment by AdsonCicilioti** [comment]
Hey @casperibo, which type field you set up for location on prisma.schema?

**Reply by casperibo** [reply]
Hi @AdsonCicilioti Sorry for the delayed response. I'm used "Json" type for location field.

```
location    Json
```

**Reply by AdsonCicilioti** [reply]
Nice! But, how do you make the '2dsphere' index on prisma.schema?

Can you show your prisma.schema?

**Reply by AdsonCicilioti** [reply]
On my case, I need to make a query to get the nearest locations. I'm using mongoose for now.

**Reply by casperibo** [reply]
You should create "2dsphere" index on MongoDB Atlas website If you have. And unfortunately you should re create the "2dsphere" index when you make "npx prisma db push" command. The command deletes the index.

**Reply by kasir-barati** [reply]
> You should create "2dsphere" index on MongoDB Atlas website If you have. And unfortunately you should re create the "2dsphere" index when you make "npx prisma db push" command. The command deletes the index.

For real? It is really painful, Any progress or simpler solution?

# Come back from the future:
You can do this: https://github.com/prisma/prisma/issues/12751#issuecomment-1093995044

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
